packages/cloud_firestore/cloud_firestore/example/lib/main.dart

```diff
@@ -9,6 +9,7 @@ import 'package:firebase_core/firebase_core.dart';
 import 'package:cloud_firestore/cloud_firestore.dart';
 
 Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
```

Comment:
Without we hit an error when Firestore is registering `MethodChannelFirestore` and setting it in `Firestore.instance`. That's because our `FirestorePlatform` has static `_instance = MethodChannelFirestore()` by default.

here's the error: 
```
Unhandled Exception: ServicesBinding.defaultBinaryMessenger was accessed before the binding was initialized.
If you're running an application and need to access the binary messenger before `runApp()` has been called (for example, during plugin initialization), then you need to explicitly call the `WidgetsFlutterBinding.ensureInitialized()` first.
If you're running a test, you can call the `TestWidgetsFlutterBinding.ensureInitialized()` as the first line in your test's `main()` method to initialize the binding.
#0      defaultBinaryMessenger.<anonymous closure> (package:flutter/src/services/binary_messenger.dart:76:7)
#1      defaultBinaryMessenger (package:flutter/src/services/binary_messenger.dart:89:4)
#2      MethodChannel.binaryMessenger (package:flutter/src/services/platform_channel.dart:140:62)
#3      MethodChannel.setMethodCallHandler (package:flutter/src/services/platform_channel.dart:368:5)
#4      new MethodChannelFirestore (package:cloud_firestore_platfor<…>
```

We will need to delay this init anyways until `Firestore.instance` is called which will solve this and another issue we are currently having in FirestoreWeb when firebase app config is passed in dart ` FirebaseApp.configure()` instead of `<script>`.

Reply:
Looks like there's no escape from this line. `FirebaseApp.configure()` calls some native functions via `MethodChannel` before `runApp` is executed. So even with lazily building `Firestore` instances the app still fails to load due to `FirebaseApp`

Reply:
So you're saying that this is a bug in `FirebaseApp` on the MethodChannel implementation? :P

Reply:
How is this different from [FirebaseAuth](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart#L17-L18), though?

Reply:
The [example for it](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_auth/firebase_auth/example/lib/main.dart) doesn't use `FirebaseApp.configure()`. We can move it some where else after `runApp()`.... or remove it and rely of `json`, `plist` and `<script>` configurations to build default firebase app.

Reply:
Ended up delaying the call to `FirebaseApp.configure()` till after runApp. 

I had a look around `firebase_core` and there isn't much we can do over there either since the MethodChannel is exchanged due to the `configure` call. So the same error would occur even if it lazily built an instance.